structopt-derive/src/spanned.rs

```diff
@@ -85,8 +89,13 @@ impl<T: AsRef<str>> AsRef<str> for Sp<T> {
 
 impl<T: ToTokens> ToTokens for Sp<T> {
     fn to_tokens(&self, stream: &mut TokenStream) {
-        let val = &self.val;
-        let quoted = quote_spanned!(self.span=> #val);
-        stream.extend(quoted);
+        // this is the simplest way out of correct ones to change span on
+        // arbitrary token tree I can come up with
+        let tt = self.val.to_token_stream().into_iter().map(|mut tt| {
+            tt.set_span(self.span.clone());
+            tt
+        });
```

Comment:
I don't get this modification, can you explain?

Reply:
Previous approach was to use `quote_spanned!(self.span => #val)` but it didn't really work. `quote_spanned!` can't really apply this span to interpolated tokens so it used `#val.span()` (pseudocode). In most cases this span turns out to be `Span::call_site`. Current approach is 100% correct and doesn't really change anything.